src/libraries/Native/build-native.sh

```diff
@@ -77,7 +77,11 @@ fi
 
 if [[ "$__TargetOS" == OSX ]]; then
     # set default OSX deployment target
-    __CMakeArgs="-DCMAKE_OSX_DEPLOYMENT_TARGET=10.13 $__CMakeArgs"
+    if [[ "$__BuildArch" == x64 ]]; then
```

Comment:
(not for this PR but just curious) Does it always eventually get overwritten (with same value) in `src/libraries/Native/Unix/CMakeLists.txt`? If so, we can delete the `if` block.

Reply:
It does not. The src/libraries/Native/Unix/CMakeLists.txt doesn't set the variable nor the underlying option for macOS x64/arm64. I actually wonder why don't we use the settings from the eng/native/configurecompiler.cmake here. I have thought that we have unified all three subrepos to use those, but I cannot see it getting included. Do you remember when you were unifying this stuff if there was a reason to not to use the configurecompiler.cmake here? Or was it being used before and some change has removed it? Maybe the fact that mono uses this stuff too was the reason?

Reply:
Ah, you are right. I was confusing `MACOS_VERSION_MIN_FLAGS` with `CMAKE_OSX_DEPLOYMENT_TARGET`.

Around the time when we were consolidating native configurations, mono builds were being added (Android/iOS etc.), so the progress stopped to avoid conflicts with other PRs. We had cleaned up most of the initial redundancies from scripts and cmake, but this part of libraires' cmake configurations was left alone because mono targets were frequently updating these files.  Now that we have relatively a stable build matrix, that work can be continued to accomplish DRY.